File: leetcode/lc-us/cpp/1310.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> xorQueries(vector<int> &arr, vector<vector<int>> &queries) {
        vector<int> nums(arr);
        for (int i = 1; i < nums.size(); i++) {
            nums[i] ^= nums[i - 1];
        }
        vector<int> ans(queries.size());
        for (int i = 0; i < ans.size(); i++) {
            if (queries[i][0] == queries[i][1]) ans[i] = arr[queries[i][0]];
            else if (queries[i][0] == 0) ans[i] = nums[queries[i][1]];
            else ans[i] = nums[queries[i][1]] ^ nums[queries[i][0] - 1];
        }
        return ans;
    }
};

File: leetcode/lc-us/py/1310.py
from typing import List


class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        n = len(arr)
        s = [0] * (n + 1)
        for i in range(1, n + 1):
            s[i] = s[i - 1] ^ arr[i - 1]
        res = []
        for q in queries:
            res.append(s[q[1] + 1] ^ s[q[0]])
        return res
